calculator/src/main/java/edu/nextstep/camp/calculator/CalculatorActivity.kt

```diff
@@ -2,48 +2,34 @@ package edu.nextstep.camp.calculator
 import androidx.appcompat.app.AppCompatActivity
 import edu.nextstep.camp.calculator.databinding.ActivityCalculatorBinding
-import edu.nextstep.camp.calculator.domain.Expression
-import edu.nextstep.camp.calculator.domain.Operator
 
-class CalculatorActivity : AppCompatActivity(), CalculatorContract.View {
-    private lateinit var binding: ActivityCalculatorBinding
-    override lateinit var presenter: CalculatorContract.Presenter
+class CalculatorActivity : AppCompatActivity(){
+
+    private val binding by lazy { ActivityCalculatorBinding.inflate(layoutInflater) }
+
+    private val calculatorViewModel by viewModels<CalculatorViewModel>()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        binding = ActivityCalculatorBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        presenter = CalculatorPresenter(this)
-
-        binding.button0.setOnClickListener { presenter.addToExpression(0) }
-        binding.button1.setOnClickListener { presenter.addToExpression(1) }
-        binding.button2.setOnClickListener { presenter.addToExpression(2) }
-        binding.button3.setOnClickListener { presenter.addToExpression(3) }
-        binding.button4.setOnClickListener { presenter.addToExpression(4) }
-        binding.button5.setOnClickListener { presenter.addToExpression(5) }
-        binding.button6.setOnClickListener { presenter.addToExpression(6) }
-        binding.button7.setOnClickListener { presenter.addToExpression(7) }
-        binding.button8.setOnClickListener { presenter.addToExpression(8) }
-        binding.button9.setOnClickListener { presenter.addToExpression(9) }
-        binding.buttonPlus.setOnClickListener { presenter.addToExpression(Operator.Plus) }
-        binding.buttonMinus.setOnClickListener { presenter.addToExpression(Operator.Minus) }
-        binding.buttonMultiply.setOnClickListener { presenter.addToExpression(Operator.Multiply) }
-        binding.buttonDivide.setOnClickListener { presenter.addToExpression(Operator.Divide) }
-        binding.buttonDelete.setOnClickListener { presenter.removeLast() }
-        binding.buttonEquals.setOnClickListener { presenter.calculate() }
-    }
 
-    override fun showExpression(expression: Expression) {
-        binding.textView.text = expression.toString()
+        initViews()
+        observeData()
     }
 
-    override fun showResult(result: Int) {
-        binding.textView.text = result.toString()
+    private fun initViews() {
+        binding.calculatorViewModel = calculatorViewModel
+        binding.lifecycleOwner = this
     }
 
-    override fun showIncompleteExpressionError() {
-        Toast.makeText(this, R.string.incomplete_expression, Toast.LENGTH_SHORT).show()
+    private fun observeData() {
+        calculatorViewModel.failInfo.observe(this) {
+            if (!it) return@observe
+            Toast.makeText(applicationContext, "완성되지 않은 수식입니다.", Toast.LENGTH_SHORT).show()
```

Comment:
```suggestion
            Toast.makeText(this, "완성되지 않은 수식입니다.", Toast.LENGTH_SHORT).show()
```
application context는 화면이 아닌 앱 전체의 라이프사이클과 관련이 되어 있어서, 앱이 완전히 종료될 때까지 남아있습니다.
해서, 해당 화면이 종료가 되더라도 application context를 사용하는 화면은 메모리 해제가 되지 않습니다.
해당 토스트는 application이 아닌 activity와 관련이 있으니, activity의 context를 활용해보면 어떨까요?